core/src/main/java/org/bitcoinj/utils/VersionTally.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2015 Ross Nicoll.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bitcoinj.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.bitcoinj.core.NetworkParameters;
+
+/**
+ * Caching counter for the block versions within a moving window. Thread
+ * safe.
+ *
+ * @see org.bitcoinj.core.NetworkParameters#getMajorityWindow()
+ */
+public class VersionTally {
+    /**
+     * Current count for each version number. In theory could use an array for
+     * this, however means any out of sequence version numbers would require a
+     * much larger array.
+     */
+    private final Map<Long, Integer> counts = new HashMap<Long, Integer>();
```

Comment:
The window isn't large. Counting each time is plenty fast enough. So why do we need this map?